store_benchmark_outputs.py

```diff
@@ -0,0 +1,69 @@
+import os
+from datetime import datetime
+
+import requests
+from google.cloud import storage
+
+
+class GoogleCloudStorage:
+    def __init__(self, bucket_name):
+        client = storage.Client()
+        self.bucket = client.get_bucket(bucket_name)
+
+    def upload_files(
+        self, output_files, directory, output_filename_prefix=None, **kwargs
+    ):
+        for output_file in output_files:
+            output_filename = (
+                f'{output_filename_prefix}_{output_file}'
+                if output_filename_prefix
+                else output_file
+            )
+
+            blob = self.bucket.blob(f'{directory or ""}/{output_filename}')
+            blob.metadata = {**kwargs}
+            blob.upload_from_filename(filename=output_file)
+
+
+def get_runner_version(runner_url):
+    response = requests.get(f'{runner_url}/status')
+    if response.status_code == 200:
+        return response.json().get('version', '').strip()
+
+    return None
+
+
+'''
+This script allows the upload of the benchmark outputs to a GCS Bucket.
+If running locally, you must specify valid Application Default Credentials (ADC):
+https://cloud.google.com/docs/authentication/production
+'''
+if __name__ == '__main__':
+    gcs_bucket_name = os.getenv('GCS_OUTPUT_BUCKET')
+    if gcs_bucket_name:
+        host = os.getenv('HOST')
+        locust_options = os.getenv('LOCUST_OPTS')
+        requests_json = os.getenv('REQUESTS_JSON')
+        output_directory = os.getenv('OUTPUT_DIRECTORY')
+        min_wait = os.getenv('USER_WAIT_TIME_MIN_SECONDS')
+        max_wait = os.getenv('USER_WAIT_TIME_MAX_SECONDS')
+        filename_prefix = os.getenv('OUTPUT_FILENAME_PREFIX')
+        runner_version = get_runner_version(host)
+        try:
+            gcs = GoogleCloudStorage(bucket_name=gcs_bucket_name)
+            gcs.upload_files(
+                output_files=('output_requests.csv', 'output_distribution.csv'),
+                directory=output_directory,
+                output_filename_prefix=filename_prefix,
+                host=host,
+                min_wait=min_wait,
+                max_wait=max_wait,
+                requests_json=requests_json,
+                locust_options=locust_options,
+                runner_version=runner_version,
+                timestamp=int(datetime.utcnow().timestamp()),
+            )
+        except Exception as ex:
+            raise Exception(f"Error occurred during file upload to GCS - {ex}")
+    else:
+        raise Exception('GCS Bucket not specified by `GCS_OUTPUT_BUCKET`')
```

Comment:
Raising a bare exception isn't good style. ValueError may be more appropriate. Also, the error message isn't clear - something like 'GCS_OUTPUT_BUCKET is None' may be clearer.